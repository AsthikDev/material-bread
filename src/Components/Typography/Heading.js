import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import BaseText from './BaseText';

const styles = StyleSheet.create({
  headingOne: {
    fontWeight: '300',
    fontSize: 96,
    lineHeight: 96,
    letterSpacing: -1.5,
  },
  headingTwo: {
    fontWeight: '300',
    fontSize: 60,
    lineHeight: 60,
    letterSpacing: -0.5,
  },
  headingThree: {
    fontSize: 48,
    lineHeight: 48,
  },
  headingFour: {
    fontSize: 34,
    lineHeight: 34,
    letterSpacing: 0.25,
  },
  headingFive: {
    fontSize: 24,
    lineHeight: 24,
  },
  headingSix: {
    fontSize: 20,
    lineHeight: 20,
    letterSpacing: 0.15,
    fontWeight: '500',
  },
});

export default class Heading extends Component {
  static propTypes = {
    type: PropTypes.number,
    children: PropTypes.node,
  };

  render() {
    const { type } = this.props;

    let textStyle;
    if (type < 1 && type > 6) {
      textStyle = styles.headingOne;
    }
    textStyle = numberToStyle(type);

    return (
      <BaseText typographyStyles={textStyle} {...this.props}>
        {this.props.children}
      </BaseText>
    );
  }
}

function numberToStyle(num) {
  const numArray = [
    styles.headingOne,
    styles.headingTwo,
    styles.headingThree,
    styles.headingFour,
    styles.headingFive,
    styles.headingSix,
  ];
  return numArray[num - 1];
}
