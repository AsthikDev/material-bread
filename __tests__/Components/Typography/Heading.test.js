import React from 'react';
import Heading from '../../../src/Components/Typography/Heading';

import renderer from 'react-test-renderer';

test('it renders H1 correctly', () => {
  const tree = renderer.create(<Heading type={1}>Heading</Heading>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it renders H2 correctly', () => {
  const tree = renderer.create(<Heading type={2}>Heading</Heading>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it renders H3 correctly', () => {
  const tree = renderer.create(<Heading type={3}>Heading</Heading>).toJSON();
  expect(tree).toMatchSnapshot();
});
test('it renders H4 correctly', () => {
  const tree = renderer.create(<Heading type={4}>Heading</Heading>).toJSON();
  expect(tree).toMatchSnapshot();
});
test('it renders H5 correctly', () => {
  const tree = renderer.create(<Heading type={5}>Heading</Heading>).toJSON();
  expect(tree).toMatchSnapshot();
});
test('it renders H6 correctly', () => {
  const tree = renderer.create(<Heading type={6}>Heading</Heading>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it renders H1 when a value for anything', () => {
  const tree = renderer.create(<Heading type={0}>Heading</Heading>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it renders H1 when a value for anything', () => {
  const tree = renderer.create(<Heading type={8}>Heading</Heading>).toJSON();
  expect(tree).toMatchSnapshot();
});
