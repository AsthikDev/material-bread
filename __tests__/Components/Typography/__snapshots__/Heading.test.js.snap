// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders H1 correctly 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0,0,0, 87)",
      "fontSize": 96,
      "fontWeight": "300",
      "letterSpacing": -1.5,
      "lineHeight": 96,
      "marginBottom": 0,
      "textAlign": "left",
    }
  }
  theme={
    Object {
      "bodyText": Object {
        "fontSize": 16,
        "letterSpacing": 0.5,
        "lineHeight": 16,
      },
      "bodyTextTwo": Object {
        "fontSize": 14,
        "letterSpacing": 0.25,
        "lineHeight": 14,
      },
      "button": Object {
        "borderRadius": 4,
        "color": "#2196f3",
        "height": 36,
        "minHeight": 36,
        "padding": 8,
      },
      "buttonText": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 1.25,
      },
      "caption": Object {
        "fontSize": 12,
        "letterSpacing": 0.4,
        "lineHeight": 12,
      },
      "containedButton": Object {
        "backgroundColor": "#2196f3",
        "paddingHorizontal": 16,
      },
      "divider": Object {
        "backgroundColor": "black",
        "height": 0.5,
        "marginVertical": 16,
        "opacity": 0.12,
        "width": "100%",
      },
      "headingFive": Object {
        "fontSize": 24,
        "lineHeight": 24,
      },
      "headingFour": Object {
        "fontSize": 34,
        "letterSpacing": 0.25,
        "lineHeight": 34,
      },
      "headingOne": Object {
        "fontSize": 96,
        "fontWeight": "300",
        "letterSpacing": -1.5,
        "lineHeight": 96,
      },
      "headingSix": Object {
        "fontSize": 20,
        "fontWeight": "500",
        "letterSpacing": 0.15,
        "lineHeight": 20,
      },
      "headingThree": Object {
        "fontSize": 48,
        "lineHeight": 48,
      },
      "headingTwo": Object {
        "fontSize": 60,
        "fontWeight": "300",
        "letterSpacing": -0.5,
        "lineHeight": 60,
      },
      "outlinedButton": Object {
        "borderColor": "rgba(0,0,0,.29)",
        "borderWidth": 0.5,
        "paddingHorizontal": 16,
      },
      "overline": Object {
        "fontSize": 10,
        "letterSpacing": 1.5,
        "lineHeight": 10,
        "textTransform": "uppercase",
      },
      "paper": Object {
        "backgroundColor": "white",
        "borderRadius": 2,
      },
      "subtitleOne": Object {
        "fontSize": 16,
        "letterSpacing": 0.15,
        "lineHeight": 16,
      },
      "subtitleTwo": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 14,
      },
    }
  }
  type={1}
  typographyStyles={
    Object {
      "fontSize": 96,
      "fontWeight": "300",
      "letterSpacing": -1.5,
      "lineHeight": 96,
    }
  }
>
  Heading
</Text>
`;

exports[`it renders H1 when a value for anything 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0,0,0, 87)",
      "marginBottom": 0,
      "textAlign": "left",
    }
  }
  theme={
    Object {
      "bodyText": Object {
        "fontSize": 16,
        "letterSpacing": 0.5,
        "lineHeight": 16,
      },
      "bodyTextTwo": Object {
        "fontSize": 14,
        "letterSpacing": 0.25,
        "lineHeight": 14,
      },
      "button": Object {
        "borderRadius": 4,
        "color": "#2196f3",
        "height": 36,
        "minHeight": 36,
        "padding": 8,
      },
      "buttonText": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 1.25,
      },
      "caption": Object {
        "fontSize": 12,
        "letterSpacing": 0.4,
        "lineHeight": 12,
      },
      "containedButton": Object {
        "backgroundColor": "#2196f3",
        "paddingHorizontal": 16,
      },
      "divider": Object {
        "backgroundColor": "black",
        "height": 0.5,
        "marginVertical": 16,
        "opacity": 0.12,
        "width": "100%",
      },
      "headingFive": Object {
        "fontSize": 24,
        "lineHeight": 24,
      },
      "headingFour": Object {
        "fontSize": 34,
        "letterSpacing": 0.25,
        "lineHeight": 34,
      },
      "headingOne": Object {
        "fontSize": 96,
        "fontWeight": "300",
        "letterSpacing": -1.5,
        "lineHeight": 96,
      },
      "headingSix": Object {
        "fontSize": 20,
        "fontWeight": "500",
        "letterSpacing": 0.15,
        "lineHeight": 20,
      },
      "headingThree": Object {
        "fontSize": 48,
        "lineHeight": 48,
      },
      "headingTwo": Object {
        "fontSize": 60,
        "fontWeight": "300",
        "letterSpacing": -0.5,
        "lineHeight": 60,
      },
      "outlinedButton": Object {
        "borderColor": "rgba(0,0,0,.29)",
        "borderWidth": 0.5,
        "paddingHorizontal": 16,
      },
      "overline": Object {
        "fontSize": 10,
        "letterSpacing": 1.5,
        "lineHeight": 10,
        "textTransform": "uppercase",
      },
      "paper": Object {
        "backgroundColor": "white",
        "borderRadius": 2,
      },
      "subtitleOne": Object {
        "fontSize": 16,
        "letterSpacing": 0.15,
        "lineHeight": 16,
      },
      "subtitleTwo": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 14,
      },
    }
  }
  type={0}
>
  Heading
</Text>
`;

exports[`it renders H1 when a value for anything 2`] = `
<Text
  style={
    Object {
      "color": "rgba(0,0,0, 87)",
      "marginBottom": 0,
      "textAlign": "left",
    }
  }
  theme={
    Object {
      "bodyText": Object {
        "fontSize": 16,
        "letterSpacing": 0.5,
        "lineHeight": 16,
      },
      "bodyTextTwo": Object {
        "fontSize": 14,
        "letterSpacing": 0.25,
        "lineHeight": 14,
      },
      "button": Object {
        "borderRadius": 4,
        "color": "#2196f3",
        "height": 36,
        "minHeight": 36,
        "padding": 8,
      },
      "buttonText": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 1.25,
      },
      "caption": Object {
        "fontSize": 12,
        "letterSpacing": 0.4,
        "lineHeight": 12,
      },
      "containedButton": Object {
        "backgroundColor": "#2196f3",
        "paddingHorizontal": 16,
      },
      "divider": Object {
        "backgroundColor": "black",
        "height": 0.5,
        "marginVertical": 16,
        "opacity": 0.12,
        "width": "100%",
      },
      "headingFive": Object {
        "fontSize": 24,
        "lineHeight": 24,
      },
      "headingFour": Object {
        "fontSize": 34,
        "letterSpacing": 0.25,
        "lineHeight": 34,
      },
      "headingOne": Object {
        "fontSize": 96,
        "fontWeight": "300",
        "letterSpacing": -1.5,
        "lineHeight": 96,
      },
      "headingSix": Object {
        "fontSize": 20,
        "fontWeight": "500",
        "letterSpacing": 0.15,
        "lineHeight": 20,
      },
      "headingThree": Object {
        "fontSize": 48,
        "lineHeight": 48,
      },
      "headingTwo": Object {
        "fontSize": 60,
        "fontWeight": "300",
        "letterSpacing": -0.5,
        "lineHeight": 60,
      },
      "outlinedButton": Object {
        "borderColor": "rgba(0,0,0,.29)",
        "borderWidth": 0.5,
        "paddingHorizontal": 16,
      },
      "overline": Object {
        "fontSize": 10,
        "letterSpacing": 1.5,
        "lineHeight": 10,
        "textTransform": "uppercase",
      },
      "paper": Object {
        "backgroundColor": "white",
        "borderRadius": 2,
      },
      "subtitleOne": Object {
        "fontSize": 16,
        "letterSpacing": 0.15,
        "lineHeight": 16,
      },
      "subtitleTwo": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 14,
      },
    }
  }
  type={8}
>
  Heading
</Text>
`;

exports[`it renders H2 correctly 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0,0,0, 87)",
      "fontSize": 60,
      "fontWeight": "300",
      "letterSpacing": -0.5,
      "lineHeight": 60,
      "marginBottom": 0,
      "textAlign": "left",
    }
  }
  theme={
    Object {
      "bodyText": Object {
        "fontSize": 16,
        "letterSpacing": 0.5,
        "lineHeight": 16,
      },
      "bodyTextTwo": Object {
        "fontSize": 14,
        "letterSpacing": 0.25,
        "lineHeight": 14,
      },
      "button": Object {
        "borderRadius": 4,
        "color": "#2196f3",
        "height": 36,
        "minHeight": 36,
        "padding": 8,
      },
      "buttonText": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 1.25,
      },
      "caption": Object {
        "fontSize": 12,
        "letterSpacing": 0.4,
        "lineHeight": 12,
      },
      "containedButton": Object {
        "backgroundColor": "#2196f3",
        "paddingHorizontal": 16,
      },
      "divider": Object {
        "backgroundColor": "black",
        "height": 0.5,
        "marginVertical": 16,
        "opacity": 0.12,
        "width": "100%",
      },
      "headingFive": Object {
        "fontSize": 24,
        "lineHeight": 24,
      },
      "headingFour": Object {
        "fontSize": 34,
        "letterSpacing": 0.25,
        "lineHeight": 34,
      },
      "headingOne": Object {
        "fontSize": 96,
        "fontWeight": "300",
        "letterSpacing": -1.5,
        "lineHeight": 96,
      },
      "headingSix": Object {
        "fontSize": 20,
        "fontWeight": "500",
        "letterSpacing": 0.15,
        "lineHeight": 20,
      },
      "headingThree": Object {
        "fontSize": 48,
        "lineHeight": 48,
      },
      "headingTwo": Object {
        "fontSize": 60,
        "fontWeight": "300",
        "letterSpacing": -0.5,
        "lineHeight": 60,
      },
      "outlinedButton": Object {
        "borderColor": "rgba(0,0,0,.29)",
        "borderWidth": 0.5,
        "paddingHorizontal": 16,
      },
      "overline": Object {
        "fontSize": 10,
        "letterSpacing": 1.5,
        "lineHeight": 10,
        "textTransform": "uppercase",
      },
      "paper": Object {
        "backgroundColor": "white",
        "borderRadius": 2,
      },
      "subtitleOne": Object {
        "fontSize": 16,
        "letterSpacing": 0.15,
        "lineHeight": 16,
      },
      "subtitleTwo": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 14,
      },
    }
  }
  type={2}
  typographyStyles={
    Object {
      "fontSize": 60,
      "fontWeight": "300",
      "letterSpacing": -0.5,
      "lineHeight": 60,
    }
  }
>
  Heading
</Text>
`;

exports[`it renders H3 correctly 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0,0,0, 87)",
      "fontSize": 48,
      "lineHeight": 48,
      "marginBottom": 0,
      "textAlign": "left",
    }
  }
  theme={
    Object {
      "bodyText": Object {
        "fontSize": 16,
        "letterSpacing": 0.5,
        "lineHeight": 16,
      },
      "bodyTextTwo": Object {
        "fontSize": 14,
        "letterSpacing": 0.25,
        "lineHeight": 14,
      },
      "button": Object {
        "borderRadius": 4,
        "color": "#2196f3",
        "height": 36,
        "minHeight": 36,
        "padding": 8,
      },
      "buttonText": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 1.25,
      },
      "caption": Object {
        "fontSize": 12,
        "letterSpacing": 0.4,
        "lineHeight": 12,
      },
      "containedButton": Object {
        "backgroundColor": "#2196f3",
        "paddingHorizontal": 16,
      },
      "divider": Object {
        "backgroundColor": "black",
        "height": 0.5,
        "marginVertical": 16,
        "opacity": 0.12,
        "width": "100%",
      },
      "headingFive": Object {
        "fontSize": 24,
        "lineHeight": 24,
      },
      "headingFour": Object {
        "fontSize": 34,
        "letterSpacing": 0.25,
        "lineHeight": 34,
      },
      "headingOne": Object {
        "fontSize": 96,
        "fontWeight": "300",
        "letterSpacing": -1.5,
        "lineHeight": 96,
      },
      "headingSix": Object {
        "fontSize": 20,
        "fontWeight": "500",
        "letterSpacing": 0.15,
        "lineHeight": 20,
      },
      "headingThree": Object {
        "fontSize": 48,
        "lineHeight": 48,
      },
      "headingTwo": Object {
        "fontSize": 60,
        "fontWeight": "300",
        "letterSpacing": -0.5,
        "lineHeight": 60,
      },
      "outlinedButton": Object {
        "borderColor": "rgba(0,0,0,.29)",
        "borderWidth": 0.5,
        "paddingHorizontal": 16,
      },
      "overline": Object {
        "fontSize": 10,
        "letterSpacing": 1.5,
        "lineHeight": 10,
        "textTransform": "uppercase",
      },
      "paper": Object {
        "backgroundColor": "white",
        "borderRadius": 2,
      },
      "subtitleOne": Object {
        "fontSize": 16,
        "letterSpacing": 0.15,
        "lineHeight": 16,
      },
      "subtitleTwo": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 14,
      },
    }
  }
  type={3}
  typographyStyles={
    Object {
      "fontSize": 48,
      "lineHeight": 48,
    }
  }
>
  Heading
</Text>
`;

exports[`it renders H4 correctly 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0,0,0, 87)",
      "fontSize": 34,
      "letterSpacing": 0.25,
      "lineHeight": 34,
      "marginBottom": 0,
      "textAlign": "left",
    }
  }
  theme={
    Object {
      "bodyText": Object {
        "fontSize": 16,
        "letterSpacing": 0.5,
        "lineHeight": 16,
      },
      "bodyTextTwo": Object {
        "fontSize": 14,
        "letterSpacing": 0.25,
        "lineHeight": 14,
      },
      "button": Object {
        "borderRadius": 4,
        "color": "#2196f3",
        "height": 36,
        "minHeight": 36,
        "padding": 8,
      },
      "buttonText": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 1.25,
      },
      "caption": Object {
        "fontSize": 12,
        "letterSpacing": 0.4,
        "lineHeight": 12,
      },
      "containedButton": Object {
        "backgroundColor": "#2196f3",
        "paddingHorizontal": 16,
      },
      "divider": Object {
        "backgroundColor": "black",
        "height": 0.5,
        "marginVertical": 16,
        "opacity": 0.12,
        "width": "100%",
      },
      "headingFive": Object {
        "fontSize": 24,
        "lineHeight": 24,
      },
      "headingFour": Object {
        "fontSize": 34,
        "letterSpacing": 0.25,
        "lineHeight": 34,
      },
      "headingOne": Object {
        "fontSize": 96,
        "fontWeight": "300",
        "letterSpacing": -1.5,
        "lineHeight": 96,
      },
      "headingSix": Object {
        "fontSize": 20,
        "fontWeight": "500",
        "letterSpacing": 0.15,
        "lineHeight": 20,
      },
      "headingThree": Object {
        "fontSize": 48,
        "lineHeight": 48,
      },
      "headingTwo": Object {
        "fontSize": 60,
        "fontWeight": "300",
        "letterSpacing": -0.5,
        "lineHeight": 60,
      },
      "outlinedButton": Object {
        "borderColor": "rgba(0,0,0,.29)",
        "borderWidth": 0.5,
        "paddingHorizontal": 16,
      },
      "overline": Object {
        "fontSize": 10,
        "letterSpacing": 1.5,
        "lineHeight": 10,
        "textTransform": "uppercase",
      },
      "paper": Object {
        "backgroundColor": "white",
        "borderRadius": 2,
      },
      "subtitleOne": Object {
        "fontSize": 16,
        "letterSpacing": 0.15,
        "lineHeight": 16,
      },
      "subtitleTwo": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 14,
      },
    }
  }
  type={4}
  typographyStyles={
    Object {
      "fontSize": 34,
      "letterSpacing": 0.25,
      "lineHeight": 34,
    }
  }
>
  Heading
</Text>
`;

exports[`it renders H5 correctly 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0,0,0, 87)",
      "fontSize": 24,
      "lineHeight": 24,
      "marginBottom": 0,
      "textAlign": "left",
    }
  }
  theme={
    Object {
      "bodyText": Object {
        "fontSize": 16,
        "letterSpacing": 0.5,
        "lineHeight": 16,
      },
      "bodyTextTwo": Object {
        "fontSize": 14,
        "letterSpacing": 0.25,
        "lineHeight": 14,
      },
      "button": Object {
        "borderRadius": 4,
        "color": "#2196f3",
        "height": 36,
        "minHeight": 36,
        "padding": 8,
      },
      "buttonText": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 1.25,
      },
      "caption": Object {
        "fontSize": 12,
        "letterSpacing": 0.4,
        "lineHeight": 12,
      },
      "containedButton": Object {
        "backgroundColor": "#2196f3",
        "paddingHorizontal": 16,
      },
      "divider": Object {
        "backgroundColor": "black",
        "height": 0.5,
        "marginVertical": 16,
        "opacity": 0.12,
        "width": "100%",
      },
      "headingFive": Object {
        "fontSize": 24,
        "lineHeight": 24,
      },
      "headingFour": Object {
        "fontSize": 34,
        "letterSpacing": 0.25,
        "lineHeight": 34,
      },
      "headingOne": Object {
        "fontSize": 96,
        "fontWeight": "300",
        "letterSpacing": -1.5,
        "lineHeight": 96,
      },
      "headingSix": Object {
        "fontSize": 20,
        "fontWeight": "500",
        "letterSpacing": 0.15,
        "lineHeight": 20,
      },
      "headingThree": Object {
        "fontSize": 48,
        "lineHeight": 48,
      },
      "headingTwo": Object {
        "fontSize": 60,
        "fontWeight": "300",
        "letterSpacing": -0.5,
        "lineHeight": 60,
      },
      "outlinedButton": Object {
        "borderColor": "rgba(0,0,0,.29)",
        "borderWidth": 0.5,
        "paddingHorizontal": 16,
      },
      "overline": Object {
        "fontSize": 10,
        "letterSpacing": 1.5,
        "lineHeight": 10,
        "textTransform": "uppercase",
      },
      "paper": Object {
        "backgroundColor": "white",
        "borderRadius": 2,
      },
      "subtitleOne": Object {
        "fontSize": 16,
        "letterSpacing": 0.15,
        "lineHeight": 16,
      },
      "subtitleTwo": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 14,
      },
    }
  }
  type={5}
  typographyStyles={
    Object {
      "fontSize": 24,
      "lineHeight": 24,
    }
  }
>
  Heading
</Text>
`;

exports[`it renders H6 correctly 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0,0,0, 87)",
      "fontSize": 20,
      "fontWeight": "500",
      "letterSpacing": 0.15,
      "lineHeight": 20,
      "marginBottom": 0,
      "textAlign": "left",
    }
  }
  theme={
    Object {
      "bodyText": Object {
        "fontSize": 16,
        "letterSpacing": 0.5,
        "lineHeight": 16,
      },
      "bodyTextTwo": Object {
        "fontSize": 14,
        "letterSpacing": 0.25,
        "lineHeight": 14,
      },
      "button": Object {
        "borderRadius": 4,
        "color": "#2196f3",
        "height": 36,
        "minHeight": 36,
        "padding": 8,
      },
      "buttonText": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 1.25,
      },
      "caption": Object {
        "fontSize": 12,
        "letterSpacing": 0.4,
        "lineHeight": 12,
      },
      "containedButton": Object {
        "backgroundColor": "#2196f3",
        "paddingHorizontal": 16,
      },
      "divider": Object {
        "backgroundColor": "black",
        "height": 0.5,
        "marginVertical": 16,
        "opacity": 0.12,
        "width": "100%",
      },
      "headingFive": Object {
        "fontSize": 24,
        "lineHeight": 24,
      },
      "headingFour": Object {
        "fontSize": 34,
        "letterSpacing": 0.25,
        "lineHeight": 34,
      },
      "headingOne": Object {
        "fontSize": 96,
        "fontWeight": "300",
        "letterSpacing": -1.5,
        "lineHeight": 96,
      },
      "headingSix": Object {
        "fontSize": 20,
        "fontWeight": "500",
        "letterSpacing": 0.15,
        "lineHeight": 20,
      },
      "headingThree": Object {
        "fontSize": 48,
        "lineHeight": 48,
      },
      "headingTwo": Object {
        "fontSize": 60,
        "fontWeight": "300",
        "letterSpacing": -0.5,
        "lineHeight": 60,
      },
      "outlinedButton": Object {
        "borderColor": "rgba(0,0,0,.29)",
        "borderWidth": 0.5,
        "paddingHorizontal": 16,
      },
      "overline": Object {
        "fontSize": 10,
        "letterSpacing": 1.5,
        "lineHeight": 10,
        "textTransform": "uppercase",
      },
      "paper": Object {
        "backgroundColor": "white",
        "borderRadius": 2,
      },
      "subtitleOne": Object {
        "fontSize": 16,
        "letterSpacing": 0.15,
        "lineHeight": 16,
      },
      "subtitleTwo": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 14,
      },
    }
  }
  type={6}
  typographyStyles={
    Object {
      "fontSize": 20,
      "fontWeight": "500",
      "letterSpacing": 0.15,
      "lineHeight": 20,
    }
  }
>
  Heading
</Text>
`;
